caringcaribou/modules/uds.py

```diff
@@ -714,6 +816,212 @@ def __dump_dids_wrapper(args):
+                    if len(found_subservices) == 0:
+                        print("\nNo Diagnostic Session Control Sub-Services were discovered\n", end=' ')
+                    else:
+                        print("\n")
+                        print("\nDiscovered Diagnostic Session Control Sub-Services:\n", end=' ')
+                        for subservice_id in found_subservices:
+                            nrc_description = NRC_NAMES.get(subservice_status[found_subservices.index(subservice_id)])
+                            print("\n0x{0:02x} : {1}".format(subservice_id, nrc_description), end=' ')
+                
+                if 17 in found_services:
+
+                    print("\n")
+                    print("\nEnumerating ECUReset Service:\n")
+
+                    found_subservices = []
+                    subservice_status = []
+
+                    for i in range(1,5):    
+
+                        extended_session(client_id, server_id, 3)
+
+                        #time.sleep(0.1)
+
+                        response = raw_send(client_id, server_id, 17, i)
+
+                        print("\rProbing ECUReset sub-service 0x{0:02x}".format(i), end="")
+
+                        if response is None:
+                            # No response received
+                            continue
+
+                        # Parse response
+                        if len(response) >= 2:
+                            response_id = response[0]
+                            response_service_id = response[1]
+                            if len(response) >= 3:
+                                status = response[2]
+                            if Iso14229_1.is_positive_response(response):
+                                found_subservices.append(i)
+                                subservice_status.append(0x00)
+                            elif response_id == Constants.NR_SI and response_service_id == 0x11 and status != NegativeResponseCodes.SUB_FUNCTION_NOT_SUPPORTED:
+                                # Any other response than "service not supported" counts
+                                found_subservices.append(i)
+                                subservice_status.append(response_service_id)
+
+                        time.sleep(timeout)
+
+                    # Print results
+                    if len(found_subservices) == 0:
+                        print("\nNo ECUReset Sub-Services were discovered.\n", end=' ')
+                    else:
+                        print("\n")
+                        print("\nDiscovered ECUReset Sub-Services:\n", end=' ')
+                        for subservice_id in found_subservices:
+                            nrc_description = NRC_NAMES.get(subservice_status[found_subservices.index(subservice_id)])
+                            print("\n0x{0:02x} : {1}".format(subservice_id, nrc_description), end=' ')
+
+                    
+
+                if 39 in found_services:
```

Comment:
`39` looks like a magic number here - I think it would be easier to read the code if the value `0x27` was stored in a variable, so one sees it is related to security access.